### ISSUE TITLE
Special treatment of the "admin" user

### ISSUE THREAD

**Issue by warpech** [Starcounter/Guidelines#24]
In many apps, a page is restricted to a specific group of users. 

For example, Calendar restricts the "Settings" page to a group "Admin (Calendar)", but does not put any user to that group.

I think that having a group-based restriction is OK, as long as there also is an easy way to have access to everything using a (properly secured) `admin` user.

I was discussing with @joozek78 that unless there is a very good reason not to, every app should regard the SystemUser named `admin` as one that has all permissions, even if they are not assigned explicitly.

Basically, `admin` would work as `root` on Linux.

@miyconst, @un-tone WDYT? 

cc @Starcounter/sc-apps

**Comment by joozek78** [comment]
I actually thought we agreed *not* to treat `admin` user differently (I remember being convinced on this by @warpech). That would encourage people to use one shared administrative account (bad idea) instead of assigning permissions to people's accounts.

Problem is, I don't have a good solution. We could use a special `SystemUserGroup`. My favorite idea so far is to add `IsAdmin` property - this has the advantage of being obvious and harder to cause accidental breaches.

---

PS
> every app should [...]

Fear not, this doesn't mean that every app has to implement it on its own, it just means it would mean this would be incorporated into Authorization Library and enabled by default / suggested

**Comment by andwah** [comment]
> every app should regard the SystemUser named admin as one that has all permissions, even if they are not assigned explicitly.

I think we should check if the user is member of the "Admin group" instead of a specific username name.

As i remember there is a "default" system administrator group that user may be members of (and also other groups can be connected to that group).  
if remember correct there is a function in SystemUser class (or system group) where you can check if a user is a member of a group using a hardcoded system admin group name.

**Comment by warpech** [comment]
> As i remember there is a "default" system administrator group that user may be members of

I cannot find such group. There is just a [SystemUserGroup](https://github.com/StarcounterApps/Simplified/blob/99888d7106a4782a45307d1cf53476d20648a556/Ring3/User/SystemUserGroup.cs) class, but that comes without any predefined groups.

> My favorite idea so far is to add `IsAdmin` property - this has the advantage of being obvious and harder to cause accidental breaches.

The `IsAdmin` property on `SystemUser` sounds good, because that might be the easiest to agree on, and the easiest to map to any data model.

Having a `SystemUser` and `SystemUserGroup` is something that apps need to agree on. Right now every app agrees on `SystemUser`, but not much more than that. SWO for example uses [SomebodyGroup](https://github.com/StarcounterApps/Simplified/blob/99888d7106a4782a45307d1cf53476d20648a556/Ring2/SomebodyGroup.cs) instead of `SystemUserGroup`.

@joozek78, @mmnosek wouldn't it introduce ambiguity that some permissions would be given in SWO through `SystemUserGroup`, while others through `SomebodyGroup`?

If that does not sound like a big deal, then my favorite option is to have a `SuperAdminSystemUserGroup : SystemUserGroup`.
